promgen/models.py

```diff
@@ -485,6 +485,22 @@ def copy_to(self, content_type, object_id):
 
         return self
 
+    # Custom logic before saving Rule to control the value of the annotation "rule":
+    # Format: annotations["rule"] = {domain}/rule/{id}
+    def save(self, *args, **kwargs):
+        with transaction.atomic():
+            if self.pk:
+                # When updating rule, we already have the primary key.
+                # Just set annotations["rule"] before saving to database.
+                self.annotations["rule"] = resolve_domain("rule-detail", pk=self.pk)
+                super().save(*args, **kwargs)
+            else:
+                # When creating a new rule, the primary key is typically not available until the instance is saved to the database.
```

Comment:
```suggestion
                # When creating a new rule, the primary key is typically not available until the
                # instance is saved to the database.
```

Please wrap your lines at 100 characters, including comments.